packages/fiber/src/shared/Canvas.native.tsx

```diff
@@ -87,42 +85,65 @@ export const Canvas = /*#__PURE__*/ React.forwardRef<View, Props>(
       setSize({ width, height })
     }, [])
 
+    const onContextCreate = React.useCallback((context: ExpoWebGLRenderingContext) => {
+      const canvasShim = {
+        width: context.drawingBufferWidth,
+        height: context.drawingBufferHeight,
+        style: {},
+        addEventListener: (() => {}) as any,
+        removeEventListener: (() => {}) as any,
+        clientHeight: context.drawingBufferHeight,
+        getContext: (() => context) as any,
+      } as HTMLCanvasElement
+
+      setCanvas(canvasShim)
+    }, [])
+
     // Execute JSX in the reconciler as a layout-effect
     React.useLayoutEffect(() => {
-      if (width > 0 && height > 0 && context) {
-        const store = render(
+      if (width > 0 && height > 0 && canvas) {
+        // Overwrite onCreated to apply RN bindings
+        const onCreated = (state: RootState) => {
+          // Bind events after creation
+          const manager = state.events.connect?.(canvas) as any
+          setBind(manager?.getEventHandlers())
```

Comment:
Maybe we can clean up events where there's something clearly labeled for spreading over a canvas. This wouldn't be encouraged for web though as React would then use synthetic events.